Project does not build with Java 16

Seeing issues here and there I tried to build the project and "no surprise" found it does not build. First step would  be to upgrade gradle to https://docs.gradle.org/7.0-milestone-3/release-notes.html.

After that we may need more changes.



```
~/github/asciidoctorj >>> ./gradlew test                                                                                                                            ±[A4●][master]
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :asciidoctorj-api:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':asciidoctorj-api:compileJava'.
> java.lang.IllegalAccessError: class org.gradle.internal.compiler.java.ClassNameCollector (in unnamed module @0x18434b8) cannot access class com.sun.tools.javac.code.Symbol$TypeSymbol (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.code to unnamed module @0x18434b8

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.8.2/userguide/command_line_interface.html#sec:command_line_warnings

```

--- a/asciidoctorj-wildfly-integration-test/build.gradle
+++ b/asciidoctorj-wildfly-integration-test/build.gradle
@@ -1,10 +1,15 @@
 jar.enabled = false
 
 ext {
-    arquillianVersion = '1.1.13.Final'
-    arquillianWildflyVersion = '2.1.1.Final'
+    arquillianVersion = '1.6.0.Final'
+    arquillianWildflyVersion = '3.0.1.Final'
 
     wildflyVersion = '23.0.0.Final'
+}
+
+repositories {
+    // Required by arquillian
+    maven { url "https://maven.repository.redhat.com/ga/" }
 }
 
 configurations {
@@ -23,7 +28,7 @@ dependencies {
     jbossmodule "org.jruby:jruby-complete:$jrubyVersion"
 
     jbossdist group: 'org.wildfly', name: 'wildfly-dist', version: wildflyVersion, ext: 'zip'
-    
+
     testImplementation "javax:javaee-api:7.0"
     testCompileOnly project(':asciidoctorj')
     testImplementation "junit:junit:$junitVersion"
@@ -85,9 +90,7 @@ test {
     systemProperty('jboss.home', jbossHome)
     systemProperty('module.path', jbossHome + "/modules" + File.pathSeparator + file('build/modules').absolutePath)
 
-    if (JavaVersion.current() == JavaVersion.VERSION_16) {
-        systemProperty('arquillian.launch', 'wildfly-java16')
-    } else if (JavaVersion.current().isJava11Compatible()) {
+    if (JavaVersion.current().isJava11Compatible()) {
         systemProperty('arquillian.launch', 'wildfly-java11')
     }
 
@@ -95,12 +98,7 @@ test {
     dependsOn createModule
     dependsOn unpackWildfly
     dependsOn configurations.jbossmodule
-
-    onlyIf {
-        JavaVersion.current() != JavaVersion.VERSION_16
-    }
 }
-
 
 
 configurations.all {

--- a/asciidoctorj-wildfly-integration-test/src/test/resources/arquillian.xml
+++ b/asciidoctorj-wildfly-integration-test/src/test/resources/arquillian.xml
@@ -29,14 +29,4 @@
     </configuration>
   </container>
 
-  <container qualifier="wildfly-java16">
-    <configuration>
-      <property name="jbossHome">${jboss.home}</property>
-      <!--<property name="javaVmArguments">-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005</property> -->
-      <property name="javaVmArguments">--add-modules java.se,java.security</property>
-      <property name="serverConfig">${jboss.server.config.file.name:standalone.xml}</property>
-      <property name="jbossArguments">${jboss.args:}</property>
-    </configuration>
-  </container>
-
 </arquillian>

